src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java

```diff
@@ -225,15 +225,14 @@ public DecodedResult deserialize(
       return result.get();
     }
     SchemaErrors.SchemaId schemaId = new SchemaErrors.SchemaId(context.getClusterId(), getSchemaIdFromRawBytes(bytes));
+    SchemaErrors.ConnectionId connectionId = new SchemaErrors.ConnectionId(context.getConnectionId());
     // Check if schema retrieval has failed recently
     if (schemaErrors.readSchemaIdByConnectionId(new SchemaErrors.ConnectionId(context.getConnectionId()), schemaId) != null) {
       return new DecodedResult(
           // If the schema fetch failed, we can't decode the data, so we just return the raw bytes.
           // We apply the encoderOnFailure function to the bytes before returning them.
           onFailure(encoderOnFailure, bytes),
-          String.valueOf(
-              schemaErrors.readSchemaIdByConnectionId(new SchemaErrors.ConnectionId(context.getConnectionId()), schemaId))
-      );
+          schemaErrors.readSchemaIdByConnectionId(connectionId, schemaId).message());
     }
```

Comment:
We really should look this up once and use the result multiple times. If we read multiple times, there's a chance that we'll get different results back.
```
var schemaId = getSchemaIdFromRawBytes(bytes);
var error = schemaErrors.readSchemaIdByConnectionId(
    contex.getConnectionId(),
    context.getClusterId(),
    schemaId
);
if (error != null) {
  return new DecodedResult(
      onFailure(encoderOnFailure, bytes),
      error.message()
  );
}
```